### PR TITLE
Restructure build workflow

### PR DESCRIPTION
**Pull request by EvaristeGalois11**
Hi, after a not so funny journey to find a solution for the sonar problem i came up with this PR.

The idea is to split the current Build workflow in two workflows:

- Build: This workflow will trigger for every push on main and for every PR. It will build the project as usual but without the sonar scan because in this stage the `SONAR_TOKEN` isn't visible. After that the workflow will upload the jacoco test coverage as an artifact.
- Sonar: This workflow will trigger after the previous workflow completes its execution. It will checkout the project from the repository that triggered the PR (this should be safe because no build is triggered and the code is used only for the sonar analysis). It will download the test coverage generated in the previous step (thanks github for making this as convoluted as possible) and launch the sonar scan. This time the SONAR_SECRET is visible and the scan should reach the holy server of SonarCloud.

As a final fun fact i have no idea if this PR will trigger the old workflow or the new ones, but I guess we'll found out soon enough lol.

Let me know what you think of this!

Bye!

### CODE REVIEW

**Comment by EvaristeGalois11** [comment]
Well at least the new Build workflow works :D

**Comment by armandino** [comment]
@EvaristeGalois11 thank you for looking into this and sorry about the not-so-fun part! :)

Can we prevent the 'Build' workflow on PRs from running until approved, like before? Otherwise there's a risk of PR spam and reaching GitHub's usage limits.

**Comment by EvaristeGalois11** [comment]
> Can we prevent the 'Build' workflow on PRs from running until approved, like before? Otherwise there's a risk of PR spam and reaching GitHub's usage limits.

Sorry @armandino i think i used the wrong words saying "every PR".

I used the same trigger as the previous workflow for the Build part:
```
on:
  push:
    branches:
      - main
  pull_request:
    types: [ opened, synchronize, reopened ]
```

So the same settings as before should apply: first time contributors need a manual approval of the workflow, everything else will run immediately. It's a project scoped setting as far as I know.

Sonar will only run if Build is completed so it shouldn't be a problem either.

Correct me if i'm wrong of course, i'm no way an expert in github actions related things.

**Review by armandino — APPROVED** [review]

**Comment by armandino** [comment]
@EvaristeGalois11 understood, thank you for clarifying :) And really appreciate your help on this!

**Comment by EvaristeGalois11** [comment]
That damn closing bracket :man_facepalming:

**Comment by armandino** [comment]
@EvaristeGalois11 no worries, it happens :) I also missed it reviewing the changes.

**Pull request merged**
